go/src/pcap/pcap.go

```diff
@@ -195,72 +238,63 @@ func handlePackets(filename string, wg *sync.WaitGroup) {
 
 	// Loop through packets in file
 	packetSource := gopacket.NewPacketSource(handle, handle.LinkType())
-	local_count := 0
 	for packet := range packetSource.Packets() {
 		//fmt.Printf("loop")
 		// We need to skip the first packet so we can calculate a timestamp
-		if local_count == 0 {
-			mut.Lock()
-			count++
-			mut.Unlock()
-			local_count++
-			previousPacket = packet
-			continue
-		}
 
 		// Increment packet counter
 		mut.Lock()
 		count++
 		mut.Unlock()
-		local_count++
 
 		// Nicely prints out which packet we are at in processing
+		mut.Lock()
 		if count%1000000 == 0 {
 			fmt.Printf("%d packets\n", count)
 		}
+		mut.Unlock()
 
 		/*********** Check for Scan ***********/
-		if packetRateGood(previousPacket, packet) {
-			// Then we get the IP information
-			// Get IPv4 Layer
-			ipLayer := packet.Layer(layers.LayerTypeIPv4)
-			var ipSrc net.IP
-			var ipDest net.IP
-			var ipId uint16
-
-			// HAS AN IP LAYER
-			if ipLayer != nil {
-				ip, _ := ipLayer.(*layers.IPv4)
-
-				//IP layer variables:
-				//Version (Either 4 or 6)
-				//IHL (IP Header Length in 32-bit words)
-				//TOS, Length, ID, Flages, FragOffset, TTL, Protocol (TCP?, etc.),
-				//Checksum, SrcIP, DstIP
-				//fmt.Printf("Source IP: %s\n", ip.SrcIP)
-				//fmt.Printf("Destin IP: %s\n", ip.DstIP)
-				//fmt.Printf("Protocol: %s\n", ip.Protocol)
-				ipId = ip.Id
-				ipSrc = ip.SrcIP
-				ipDest = ip.DstIP
-			}
-
-			tcpLayer := packet.Layer(layers.LayerTypeTCP)
-
-			// Get Destination port from TCP layer
-			if tcpLayer != nil {
-				tcp, _ := tcpLayer.(*layers.TCP)
-				var dstTCPPort = tcp.DstPort
-
-				/******** zMap Check *********/
-				checkZMap(ipSrc, dstTCPPort, ipId)
-				checkMasscan(ipSrc, ipDest, dstTCPPort, ipId, tcp.Seq)
-			}
+		// Then we get the IP information
+		// Get IPv4 Layer
+		ipLayer := packet.Layer(layers.LayerTypeIPv4)
+		var ipSrc net.IP
+		var ipDest net.IP
+		//var ipId uint16
+
+		// HAS AN IP LAYER
+		if ipLayer != nil {
+			ip, _ := ipLayer.(*layers.IPv4)
+
+			//IP layer variables:
+			//Version (Either 4 or 6)
+			//IHL (IP Header Length in 32-bit words)
+			//TOS, Length, ID, Flages, FragOffset, TTL, Protocol (TCP?, etc.),
+			//Checksum, SrcIP, DstIP
+			//fmt.Printf("Source IP: %s\n", ip.SrcIP)
+			//fmt.Printf("Destin IP: %s\n", ip.DstIP)
+			//fmt.Printf("Protocol: %s\n", ip.Protocol)
+			//ipId = ip.Id
+			ipSrc = ip.SrcIP
+			ipDest = ip.DstIP
+
 		} else {
-			fmt.Println("Skipping...")
+			fmt.Println("I didn't want this packet anyways")
```

Comment:
same

Reply:
In 240 million packets there were like 400 that got skipped